src/Build/BackEnd/Node/OutOfProcNode.cs

```diff
@@ -476,20 +476,9 @@ private NodeEngineShutdownReason HandleShutdown(out Exception exception)
             // If the node was never configured, this will be null.
             if (_savedEnvironment != null)
             {
-                foreach (KeyValuePair<string, string> entry in CommunicationsUtilities.GetEnvironmentVariables())
-                {
-                    if (!_savedEnvironment.ContainsKey(entry.Key))
-                    {
-                        Environment.SetEnvironmentVariable(entry.Key, null);
-                    }
-                }
-
-                foreach (KeyValuePair<string, string> entry in _savedEnvironment)
-                {
-                    Environment.SetEnvironmentVariable(entry.Key, entry.Value);
-                }
+                CommunicationsUtilities.SetEnvironment(_savedEnvironment);
+                Traits.UpdateFromEnvironment();
```

Comment:
How about adding `bool updateTraits` argument to the `SetEnvironment` to avoid the need to repetitive calls (with exception of InProcNode)?

Reply:
Apparently, pattern varies a bit so I sometimes can use `SetEnvironment` and sometimes not and sometimes I need to update traits and sometimes do not. So, I decided to keep them separate for time being.